allauth/socialaccount/providers/oauth2/views.py

```diff
@@ -64,12 +64,17 @@ def view(request, *args, **kwargs):
                 return e.response
         return view
 
-    def get_client(self, request, app):
+    def get_callback_url(self, request, app):
```

Comment:
This particular change is not related to the BattleNetOAuth2Adapter - see my comments above. I left it in because I think it improves the code.